Userland/Libraries/LibJS/JIT/Compiler.cpp

```diff
@@ -2722,7 +2729,15 @@ OwnPtr<NativeExecutable> Compiler::compile(Bytecode::Executable& bytecode_execut
         dbgln("\033[32;1mJIT compilation succeeded!\033[0m {}", bytecode_executable.name);
     }
 
-    auto executable = make<NativeExecutable>(executable_memory, compiler.m_output.size(), mapping);
+    // Build executable
+    auto const code = ReadonlyBytes {
+        executable_memory,
+        compiler.m_output.size(),
+    };
+
+    auto gdb_object = build_gdb_image(code);
```

Comment:
Creating an ELF for every single function is probably quite expensive and should be behind a flag, so it's only done when needed for debugging.